src/main/java/de/braintags/io/vertx/util/asyncLock/AsyncReadWriteLock.java

```diff
@@ -0,0 +1,145 @@
+package de.braintags.io.vertx.util.asyncLock;

```

Comment:
In general: really dark magic :)
Please open discussion under vertx-developer or vertx to check potential side affects